### ISSUE TITLE
If the RemoteApp alias has a space in it, RemoteApp Manager will crash

### ISSUE THREAD

**Issue by PeterStrick** [kimmknight/remoteapptool#37]
RemoteApp Manager will crash when the Properties of a RemoteApp with a space in its alias get opened.

Example:
RemoteApp Alias: "Dashboard" opens its properties fine.
RemoteApp Alias: "Dashboard " crashes RemoteApp Manager

Ex. 2:
RemoteApp Alias: "Excel" opens its properties fine.
RemoteApp Alias: "Excel (2)" crashes the RemoteApp alias.

System configuration:
Windows Server 2012R2 Essentials x64 with Remotedesktop-Roles installed.

The RemoteApps are being opened using the RDWeb.

Screenshots are provided. (German)
![image](https://user-images.githubusercontent.com/60312421/92757271-dfdd9300-f38d-11ea-88bc-5ec7401a266c.png)

Complete Error log: (German)
```log
Informationen über das Aufrufen von JIT-Debuggen
anstelle dieses Dialogfelds finden Sie am Ende dieser Meldung.

************** Ausnahmetext **************
System.ArgumentOutOfRangeException: InvalidArgument=Value mit dem Wert -1 ist für start ungültig.
Parametername: start
   bei System.Windows.Forms.TextBoxBase.Select(Int32 start, Int32 length)
   bei RemoteApp_Tool.RemoteAppFunction.ValidateTextBoxR(TextBox TheTextBox, String regex)
   bei RemoteApp_Tool.RemoteAppFunction.ValidateAppName(TextBox TheTextBox)
   bei RemoteApp_Tool.RemoteAppEditWindow.ShortNameText_TextChanged(Object sender, EventArgs e)
   bei System.Windows.Forms.Control.OnTextChanged(EventArgs e)
   bei System.Windows.Forms.Control.set_Text(String value)
   bei System.Windows.Forms.TextBoxBase.set_Text(String value)
   bei System.Windows.Forms.TextBox.set_Text(String value)
   bei RemoteApp_Tool.RemoteAppEditWindow.LoadValues()
   bei RemoteApp_Tool.RemoteAppEditWindow.EditRemoteApp(RemoteApp SelectedRemoteApp)
   bei RemoteApp_Tool.RemoteAppMainWindow.EditRemoteApp(String AppName)
   bei RemoteApp_Tool.RemoteAppMainWindow.EditButton_Click(Object sender, EventArgs e)
   bei System.Windows.Forms.Control.OnClick(EventArgs e)
   bei System.Windows.Forms.Button.OnClick(EventArgs e)
   bei System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   bei System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   bei System.Windows.Forms.Control.WndProc(Message& m)
   bei System.Windows.Forms.ButtonBase.WndProc(Message& m)
   bei System.Windows.Forms.Button.WndProc(Message& m)
   bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Geladene Assemblys **************
mscorlib
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll.
----------------------------------------
RemoteApp Tool
    Assembly-Version: 5.3.0.0.
    Win32-Version: 5.3.0.0.
    CodeBase: file:///C:/Program%20Files%20(x86)/Windows%20Server/Tools/RemoteApp/RemoteApp%20Tool.exe.
----------------------------------------
Microsoft.VisualBasic
    Assembly-Version: 10.0.0.0.
    Win32-Version: 14.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualBasic.dll.
----------------------------------------
System
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll.
----------------------------------------
System.Core
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll.
----------------------------------------
System.Windows.Forms
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll.
----------------------------------------
System.Drawing
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll.
----------------------------------------
System.Configuration
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll.
----------------------------------------
System.Xml
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll.
----------------------------------------
System.Runtime.Remoting
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.3761.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Remoting/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Remoting.dll.
----------------------------------------
RemoteAppLib
    Assembly-Version: 1.0.0.1.
    Win32-Version: 1.0.0.1.
    CodeBase: file:///C:/Program%20Files%20(x86)/Windows%20Server/Tools/RemoteApp/RemoteAppLib.DLL.
----------------------------------------
IconLib
    Assembly-Version: 0.73.0.0.
    Win32-Version: 0.73.0.0.
    CodeBase: file:///C:/Program%20Files%20(x86)/Windows%20Server/Tools/RemoteApp/IconLib.DLL.
----------------------------------------
mscorlib.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.7.3062.0 built by: NET472REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_de_b77a5c561934e089/mscorlib.resources.dll.
----------------------------------------
Microsoft.VisualBasic.resources
    Assembly-Version: 10.0.0.0.
    Win32-Version: 14.7.3062.0 built by: NET472REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic.resources/v4.0_10.0.0.0_de_b03f5f7f11d50a3a/Microsoft.VisualBasic.resources.dll.
----------------------------------------
System.Windows.Forms.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.7.3062.0 built by: NET472REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_de_b77a5c561934e089/System.Windows.Forms.resources.dll.
----------------------------------------

************** JIT-Debuggen **************
Um das JIT-Debuggen (Just-In-Time) zu aktivieren, muss in der
Konfigurationsdatei der Anwendung oder des Computers
(machine.config) der jitDebugging-Wert im Abschnitt system.windows.forms festgelegt werden.
Die Anwendung muss mit aktiviertem Debuggen kompiliert werden.

Zum Beispiel:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

Wenn das JIT-Debuggen aktiviert ist, werden alle nicht behandelten
Ausnahmen an den JIT-Debugger gesendet, der auf dem
Computer registriert ist, und nicht in diesem Dialogfeld behandelt.
```

**Comment by kimmknight** [comment]
Thank you for this. Yes, I can confirm this is an issue with the latest RemoteApp Tool build. I'll leave the issue open to investigate further.

For the meantime, if you want to use RemoteApp Tool, you might have to rename the remoteapps using Microsoft's tools, or rename them in regedit under: 
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Terminal Server\TSAppAllowList\Applications

**Comment by wonleing** [comment]
how's going on this issue? Is this a fork project or THE upstream?

**Comment by MrBrianGale** [comment]
I may be mistaken, but I think this is fixed in version 6.0.0.0.
This is the upstream - Kimmknight made the original tool and put it on here.

I was just trying to reproduce this in version 6 and I am not able to, but it COULD be I am not following the same steps as the OP.

**Comment by PeterStrick** [comment]
> I may be mistaken, but I think this is fixed in version 6.0.0.0. This is the upstream - Kimmknight made the original tool and put it on here.
> 
> I was just trying to reproduce this in version 6 and I am not able to, but it COULD be I am not following the same steps as the OP.

Well as I don´t own a Windows Server anymore, I can´t retest this issue. If it wasn´t reproducable in Version 6 then I´ll just close this issue.
